### ISSUE TITLE
Unable to build with Xcode 12.2 using -G Xcode

### ISSUE THREAD

**Issue by clanmills** [Exiv2/exiv2#1408]
Xcode 12.2 with -G Xcode does not work.  Xcode 12.2 with -G "Unix Makefiles" works.

There are several different issues.  I suspect Xcode 12.2 introduces support for the M-1 chip which is designated -arch arm64.  There are several problems:

1. CMAKE_CXX_COMPILER is not correctly set
```cmake
# Xcode 12.2 reports "No CMAKE_CXX_COMPILER could be found." on project command
if ( "${CMAKE_GENERATOR}" STREQUAL "Xcode" AND "${CMAKE_CXX_COMPILER}" STREQUAL "" )
   # execute_process(COMMAND xcrun --find c++ OUTPUT_VARIABLE CMAKE_CXX_COMPILER)
   # string(FIND "${CMAKE_CXX_COMPILER}" c++ clang CMAKE_CXX_COMPILER)
   set(CMAKE_CXX_COMPILER /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang)
endif()
```

2. execute_process is behaving strangely
xcrun --find c++ reports a link /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ and cmake says "not a valid compiler" (or something like that).  The only way I can get this to work is to hard wire the code

3. the project file exiv2.xcodeproj/project.pbxproj has references to arm64

I've tried to fix that by specifying the ARCHITECTURE in cmake/mainSetup.cmake:

```bash
554 rmills@rmillsmbp:~/gnu/github/exiv2/0.27-maintenance $ git diff cmake/mainSetup.cmake 
diff --git a/cmake/mainSetup.cmake b/cmake/mainSetup.cmake
index fcaa21fa..4517cd64 100644
--- a/cmake/mainSetup.cmake
+++ b/cmake/mainSetup.cmake
@@ -22,7 +22,10 @@ endif()
 if (UNIX)
     if (APPLE)
         set(CMAKE_MACOSX_RPATH ON)
-        set(CMAKE_INSTALL_RPATH "@loader_path")
+        set(CMAKE_INSTALL_RPATH "@loader_path"   )
+        set(CMAKE_OSX_DEPLOYMENT_TARGET "10.11.0")
+        set(CMAKE_OSX_ARCHITECTURES     "x86_64" )
+        set(CMAKE_XCODE_ARCHS           "x86_64" )
     else()
         join_paths(CMAKE_INSTALL_RPATH "$ORIGIN" ".." "${CMAKE_INSTALL_LIBDIR}")
     endif()
555 rmills@rmillsmbp:~/gnu/github/exiv2/0.27-maintenance $ 
```
4. It's setting Deployment target to 11.0 

5. The library won't link due to the arm64 code in exiv2.xcodeproj/project.pbxproj

**Comment by clanmills** [comment]
I'm unassigning myself and the v0.27.4 milestone.  There is no action here.  It has been documented.   I'm leaving this open because I think Xcode 12.2 might be OK with the M1 processor and therefore this behaviour only applies to Intel processors.

Lizzie _(the cat)_ threw up on my 7 year old MacBook Pro.  I've ordered a new MBP with the M1 Processor.  As I want a US Keyboard, it's not available immediately and will be here for Christmas.  I'll update this issue if I discover that Xcode 12.2 works correctly with the M1 processor.

**Comment by clanmills** [comment]
Xcode doesn't work correctly with the M1 processor. The machine was ordered in November and finally arrived on February 3.

Apple has replaced the function keys with a glass ribbon thing.  No function keys.  How do you debug in Xcode?

**Comment by clanmills** [comment]
I'm still having trouble with -G Xcode on M1.  It's confused about the architecture.  Some part of the build is being built with STANDARD_ARCHS which includes arm64 and x64_64.  Grrrrr.  I've been able to fix that with:

```bash
$ cmake .. -G Xcode -DCMAKE_OSX_ARCHITECTURES=arm64
```

I think we're in for a slightly bumpy ride with the M1.  Will be able to build "fat" libraries for the Mac.  Time will tell.

Xcode 12.4 is working well on the M1 processor with -G Unix Makefiles.

**About building with Apple Silicon**  

I made the error of using the Migration Assistant to clone my old MacBook Pro to the new M1 powered MacBool Pro.  This brings a load of intel build libraries.  Start fresh.  Build zlib, expat and get text from source.  Everything's good.
